test/test_jit.py

```diff
@@ -5952,84 +5952,86 @@ def func(a, b):
+        int_vals = list(range(-5, 5, 1)) + [mx_int + 5, mx_int * 2, mn_int - 5, mn_int * 2]
 
-        def test_exp_int(x):
-            # type: (int) -> float
-            return math.exp(x)
+        def checkMath(func_name, num_args, is_float=True, ret_type="float", debug=False, vals=None, args_type=None):
+            funcs_template = dedent('''
+            def func(a, b):
+                # type: {args_type} -> {ret_type}
+                return math.{func}({args})
+            ''')
+            if num_args == 1:
+                args = "a"
+            elif num_args == 2:
+                args = "a, b"
+            else:
+                raise RuntimeError("Test doesn't support more than 2 arguments")
+            if args_type is None:
+                args_type = "(float, float)" if is_float else "(int, int)"
+            funcs_str = funcs_template.format(func=func_name, args=args, args_type=args_type, ret_type=ret_type)
+            scope = {}
+            execWrapper(funcs_str, globals(), scope)
+            cu = torch.jit.CompilationUnit(funcs_str)
+            fs = cu.func
+            f = scope['func']
 
-        def test_exp_float(x):
-            # type: (float) -> float
-            return math.exp(x)
 
-        def test_sqrt_int(x):
-            # type: (int) -> float
-            return math.sqrt(x)
+            if vals is None:
+                vals = float_vals if is_float else int_vals
+                vals = [(i, j) for i in vals for j in vals]
 
-        def test_sqrt_float(x):
-            # type: (float) -> float
-            return math.sqrt(x)
+            for a, b in vals:
+                resf = None
+                resfs = None
+                try:
+                    resf = f(a, b)
+                except Exception as e:
+                    resf = e
+                try:
+                    resfs = fs(a, b)
+                except Exception as e:
+                    resfs = e
+                if debug:
+                    print("in: ", a, b)
+                    print("out: ", resf, resfs)
+                if resf != resfs:
+                    if isinstance(resf, Exception):
+                        continue
+                    if type(resf) == type(resfs):
+                        if isinstance(resf, tuple) and (math.isnan(resf[0]) == math.isnan(resfs[0])):
+                            continue
+                        if isinstance(resf, float) and math.isnan(resf) and math.isnan(resfs):
+                            continue
+                        if isinstance(resf, float) and abs(resf - resfs) < 1e-4:
+                            continue
+                    raise AssertionError("Failed on {func_name} with inputs {a} {b}. Python: {resf}, Script: {resfs}"
+                                         .format(func_name=func_name, a=a, b=b, resf=resf, resfs=resfs))
+
+
+        unary_float_ops = ["log", "log1p", "log10", "exp", "sqrt", "gamma", "lgamma", "erf", "erfc", "expm1", "fabs"]
```

Comment:
We should have something like `[x for x in dir(math) if callable(getattr(math, x))]` so we know we're getting full coverage of the `math` module instead of maintaining big lists like these (though we still need some list to say what the args are for the non-unary ones)

Reply:
I'd prefer to do this after this stack lol - otherwise I'd need to maintain a list of what functions aren't supported that I whittle down for each commit in this stack.